packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -690,6 +690,15 @@ export class Function extends FunctionBase {
     }
     this._architecture = props.architecture ?? (props.architectures && props.architectures[0]);
 
+    if (props.functionName && !Token.isUnresolved(props.functionName)) {
+      if (props.functionName.length > 140) {
```

Comment:
[this](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-functionname) says that the length cannot be greater than 64 characters. Where are you gettintg the 140 from?

Reply:
perhaps the 140 comes from here? @Dzhuneyt do you think you can help investigate the correct length here? #18795

Reply:
I've confirmed that the number is 64 here. Going to update this PR accordingly.